wrappers/python/performance/llama_perf.py

```diff
@@ -0,0 +1,251 @@
+from nntile.tensor import TensorMoments, TensorTraits
+
+# Create argument parser
+parser = argparse.ArgumentParser(prog="Test performance script for LLaMa",
+        description="This example presents an NNTile implementation of a "
+        "LLaMa-family of models and compares it against the Huggingface. "
+        "It checks relative accuracy of a forward pass (values of "
+        "activations) and backward pass (gradients of parameters and "
+        "activations) and a throughput of inference and training. It can "
+        "also fine-tune a pretrained NNTile model on a chosen dataset.")
+
+parser.add_argument("--config-path", type=str, default="")
+parser.add_argument("--submodule", choices=["mlp", "decoder", "causal_llama"],
+                    default="mlp")
+
+parser.add_argument("--seq-len", type=int, default=1024)
+parser.add_argument("--seq-len-tile", type=int, default=-1)
+parser.add_argument("--minibatch-size", type=int, default=1)
+parser.add_argument("--minibatch-size-tile", type=int, default=-1)
+
+parser.add_argument("--hidden-size-tile", type=int, default=-1)
+parser.add_argument("--intermediate-size-tile", type=int, default=-1)
+parser.add_argument("--n-head-tile", type=int, default=-1)
+
+parser.add_argument("--dtype", choices=["fp32", "fp32_fast_tf32", "bf16",
+                                        "fp32_fast_fp16", "fp32_fast_bf16"],
+                    default="fp32")
+parser.add_argument("--restrict", choices=["cpu", "cuda", None],
+                    default=None)
+parser.add_argument("--flash-attention", action="store_true")
+parser.add_argument("--use-redux", action="store_true")
+parser.add_argument("--torch-compile", action="store_true")
+parser.add_argument("--num_warmup_calls", type=int, default=1)
+parser.add_argument("--logger", action="store_true")
+parser.add_argument("--logger-server-addr", type=str,
+                    default="localhost")
+parser.add_argument("--logger-server-port", type=int, default=5001)
+
+# Parse arguments
+args = parser.parse_args()
+print(args)
+
+if args.seq_len_tile == -1:
+    args.seq_len_tile = args.seq_len
+if args.minibatch_size_tile == -1:
+    args.minibatch_size_tile = args.minibatch_size
+# Check arguments
+assert args.seq_len_tile > 0
+assert args.minibatch_size > 0
+assert args.minibatch_size_tile > 0
+assert args.minibatch_size % args.minibatch_size_tile == 0
+
+dtype2nntile = {
+        'fp32': nntile.tensor.Tensor_fp32,
+        'fp32_fast_tf32': nntile.tensor.Tensor_fp32_fast_tf32,
+        'bf16': nntile.tensor.Tensor_bf16,
+        'fp32_fast_fp16': nntile.tensor.Tensor_fp32_fast_fp16,
+        'fp32_fast_bf16': nntile.tensor.Tensor_fp32_fast_bf16,
+}
+
+
+f = open(args.config_path)
+conf_dict = json.load(f)
+f.close()
+llama_torch_config = LlamaConfig(**conf_dict)
+
+# Initialize NNTile and StarPU
+time0 = time.time()
+# Set up StarPU+MPI and init codelets
+nntile_config = nntile.starpu.Config(-1, -1, 1, args.logger,
```

Comment:
nntile is an option. We shall not initialise it unless it is required to run nntile perf tests.